tests/posix/common/src/fnmatch.c

```diff
@@ -4,7 +4,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <zephyr/posix/fnmatch.h>
```

Comment:
Do we have to fix all of such includes eventually?
```
grep -nr '#include <zephyr\/posix\/.*'
./drivers/wifi/simplelink/simplelink_sockets.c:12:#include <zephyr/posix/fcntl.h>
./drivers/fpga/fpga_ice40.c:21:#include <zephyr/posix/time.h>
./drivers/modem/ublox-sara-r4.c:18:#include <zephyr/posix/fcntl.h>
./include/zephyr/posix/time.h:62:#include <zephyr/posix/signal.h>
./include/zephyr/posix/semaphore.h:9:#include <zephyr/posix/time.h>
./include/zephyr/posix/mqueue.h:11:#include <zephyr/posix/time.h>
./include/zephyr/posix/mqueue.h:12:#include <zephyr/posix/fcntl.h>
./include/zephyr/posix/mqueue.h:13:#include <zephyr/posix/sys/stat.h>
./include/zephyr/posix/unistd.h:10:#include <zephyr/posix/sys/stat.h>
./include/zephyr/posix/arpa/inet.h:11:#include <zephyr/posix/netinet/in.h>
./include/zephyr/posix/arpa/inet.h:12:#include <zephyr/posix/sys/socket.h>
./include/zephyr/posix/sys/eventfd.h:13:#include <zephyr/posix/fcntl.h>
./include/zephyr/posix/pthread.h:16:#include <zephyr/posix/time.h>
./include/zephyr/posix/pthread.h:17:#include <zephyr/posix/unistd.h>
./include/zephyr/posix/pthread.h:18:#include <zephyr/posix/sched.h>
./subsys/mgmt/mcumgr/transport/src/smp_udp.c:17:#include <zephyr/posix/unistd.h>
./subsys/mgmt/mcumgr/transport/src/smp_udp.c:18:#include <zephyr/posix/sys/socket.h>
./subsys/net/lib/websocket/websocket.c:26:#include <zephyr/posix/unistd.h>
./subsys/net/lib/websocket/websocket.c:27:#include <zephyr/posix/sys/socket.h>
./subsys/net/lib/sockets/sockets_packet.c:12:#include <zephyr/posix/fcntl.h>
./subsys/net/lib/sockets/sockets.c:20:#include <zephyr/posix/fcntl.h>
./subsys/net/lib/sockets/sockets_tls.c:12:#include <zephyr/posix/fcntl.h>
./subsys/net/lib/sockets/sockets_net_mgmt.c:11:#include <zephyr/posix/fcntl.h>
./subsys/net/lib/sockets/sockets_can.c:12:#include <zephyr/posix/fcntl.h>
./subsys/net/lib/sockets/socketpair.c:9:#include <zephyr/posix/fcntl.h>
./subsys/net/lib/config/init_clock_sntp.c:12:#include <zephyr/posix/time.h>
./subsys/net/lib/mqtt_sn/mqtt_sn_transport_udp.c:21:#include <zephyr/posix/fcntl.h>
./subsys/net/lib/lwm2m/lwm2m_engine.c:38:#include <zephyr/posix/fcntl.h>
./subsys/shell/modules/devmem_service.c:16:#include <zephyr/posix/unistd.h>
./subsys/shell/modules/date_service.c:11:#include <zephyr/posix/time.h>
./subsys/shell/shell_wildcard.c:8:#include <zephyr/posix/fnmatch.h>
./tests/posix/getopt/src/main.c:18:#include <zephyr/posix/unistd.h>
./tests/posix/eventfd/src/_main.h:14:#include <zephyr/posix/poll.h>
./tests/posix/eventfd/src/_main.h:15:#include <zephyr/posix/sys/eventfd.h>
./tests/posix/eventfd/src/_main.h:16:#include <zephyr/posix/unistd.h>
./tests/posix/common/src/fnmatch.c:7:#include <zephyr/posix/fnmatch.h>
./tests/posix/common/src/clock.c:7:#include <zephyr/posix/time.h>
./tests/posix/common/src/clock.c:8:#include <zephyr/posix/sys/time.h>
./tests/posix/common/src/clock.c:9:#include <zephyr/posix/unistd.h>
./tests/posix/common/src/sleep.c:7:#include <zephyr/posix/unistd.h>
./tests/posix/common/src/nanosleep.c:9:#include <zephyr/posix/time.h>
./tests/posix/headers/src/sys_ioctl_h.c:12:#include <zephyr/posix/sys/ioctl.h>
./tests/posix/headers/src/signal_h.c:12:#include <zephyr/posix/signal.h>
./tests/posix/headers/src/net_if_h.c:12:#include <zephyr/posix/net/if.h>
./tests/posix/headers/src/sys_socket_h.c:12:#include <zephyr/posix/sys/socket.h>
./tests/posix/headers/src/sys_eventfd_h.c:12:#include <zephyr/posix/sys/eventfd.h>
./tests/posix/headers/src/unistd_h.c:12:#include <zephyr/posix/unistd.h>
./tests/posix/headers/src/semaphore_h.c:12:#include <zephyr/posix/semaphore.h>
./tests/posix/headers/src/netinet_tcp_h.c:12:#include <zephyr/posix/netinet/tcp.h>
./tests/posix/headers/src/netdb_h.c:12:#include <zephyr/posix/netdb.h>
./tests/posix/headers/src/sched_h.c:12:#include <zephyr/posix/sched.h>
./tests/posix/headers/src/sys_select_h.c:12:#include <zephyr/posix/sys/select.h>
./tests/posix/headers/src/dirent_h.c:12:#include <zephyr/posix/dirent.h>
./tests/posix/headers/src/pthread_h.c:12:#include <zephyr/posix/pthread.h>
./tests/posix/headers/src/poll_h.c:12:#include <zephyr/posix/poll.h>
./tests/posix/headers/src/arpa_inet_h.c:12:#include <zephyr/posix/arpa/inet.h>
./tests/posix/headers/src/mqueue_h.c:12:#include <zephyr/posix/mqueue.h>
./tests/posix/headers/src/netinet_in_h.c:12:#include <zephyr/posix/netinet/in.h>
./tests/posix/headers/src/sys_time_h.c:12:#include <zephyr/posix/sys/time.h>
./tests/posix/fs/src/test_fs_file.c:9:#include <zephyr/posix/unistd.h>
./tests/posix/fs/src/test_fs_stat.c:8:#include <zephyr/posix/fcntl.h>
./tests/posix/fs/src/test_fs_stat.c:9:#include <zephyr/posix/unistd.h>
./tests/posix/fs/src/test_fs_stat.c:10:#include <zephyr/posix/dirent.h>
./tests/posix/fs/src/test_fs_open_flags.c:9:#include <zephyr/posix/unistd.h>
./tests/posix/fs/src/test_fs_dir.c:9:#include <zephyr/posix/unistd.h>
./tests/posix/fs/src/test_fs_dir.c:10:#include <zephyr/posix/dirent.h>
./tests/net/socket/socketpair/src/_main.h:15:#include <zephyr/posix/fcntl.h>
./tests/net/socket/tls_ext/src/main.c:12:#include <zephyr/posix/unistd.h>
./samples/net/mdns_responder/src/service.c:10:#include <zephyr/posix/netinet/in.h>
./samples/net/cloud/aws_iot_mqtt/src/main.c:21:#include <zephyr/posix/time.h>
./samples/shields/npm6001_ek/src/main.c:14:#include <zephyr/posix/unistd.h>
./samples/subsys/shell/shell_module/src/main.c:19:#include <zephyr/posix/unistd.h>
./lib/posix/pthread_key.c:9:#include <zephyr/posix/pthread.h>
./lib/posix/pthread_key.c:10:#include <zephyr/posix/pthread_key.h>
./lib/posix/timer.c:10:#include <zephyr/posix/time.h>
./lib/posix/fnmatch.c:46:#include <zephyr/posix/fnmatch.h>
./lib/posix/pthread.c:15:#include <zephyr/posix/pthread.h>
./lib/posix/getopt/getopt.c:36:#include <zephyr/posix/unistd.h>
./lib/posix/pthread_barrier.c:11:#include <zephyr/posix/pthread.h>
./lib/posix/pthread_sched.h:12:#include <zephyr/posix/sched.h>
./lib/posix/pthread_cond.c:11:#include <zephyr/posix/pthread.h>
./lib/posix/pthread_rwlock.c:8:#include <zephyr/posix/time.h>
./lib/posix/pthread_rwlock.c:9:#include <zephyr/posix/posix_types.h>
./lib/posix/fs.c:10:#include <zephyr/posix/unistd.h>
./lib/posix/fs.c:11:#include <zephyr/posix/dirent.h>
./lib/posix/fs.c:14:#include <zephyr/posix/sys/stat.h>
./lib/posix/fs.c:15:#include <zephyr/posix/fcntl.h>
./lib/posix/clock.c:8:#include <zephyr/posix/time.h>
./lib/posix/clock.c:9:#include <zephyr/posix/sys/time.h>
./lib/posix/posix_internal.h:14:#include <zephyr/posix/pthread.h>
./lib/posix/pthread_spinlock.c:11:#include <zephyr/posix/pthread.h>
./lib/posix/sleep.c:10:#include <zephyr/posix/unistd.h>
./lib/posix/pthread_common.c:10:#include <zephyr/posix/time.h>
./lib/posix/semaphore.c:8:#include <zephyr/posix/pthread.h>
./lib/posix/mqueue.c:10:#include <zephyr/posix/time.h>
./lib/posix/mqueue.c:11:#include <zephyr/posix/mqueue.h>
./lib/posix/pthread_mutex.c:11:#include <zephyr/posix/pthread.h>
./lib/posix/pthread_sched.c:10:#include <zephyr/posix/sched.h>
./lib/posix/nanosleep.c:12:#include <zephyr/posix/time.h>
./lib/posix/signal.c:10:#include <zephyr/posix/signal.h>
./lib/posix/eventfd.c:11:#include <zephyr/posix/sys/eventfd.h>
./lib/libc/picolibc/libc-hooks.c:10:#include <zephyr/posix/sys/stat.h>
./lib/libc/newlib/libc-hooks.c:12:#include <zephyr/posix/sys/stat.h>
./lib/libc/minimal/include/fcntl.h:12:          "Please use CONFIG_POSIX_API or #include <zephyr/posix/fcntl.h>")
./lib/libc/minimal/include/fcntl.h:14:#include <zephyr/posix/fcntl.h>
./lib/libc/minimal/include/sys/stat.h:12:               "Please use CONFIG_POSIX_API or #include <zephyr/posix/sys/stat.h>")
./lib/libc/minimal/include/sys/stat.h:14:#include <zephyr/posix/sys/stat.h>
./lib/libc/common/source/time/time.c:10:#include <zephyr/posix/time.h>
./lib/os/crc_shell.c:14:#include <zephyr/posix/unistd.h>
./lib/os/fdtable.c:17:#include <zephyr/posix/fcntl.h>
./doc/releases/release-notes-3.3.rst:2428:    * I.e. ``#include <unistd.h>`` instead of ``#include <zephyr/posix/unistd.h>``.
./modules/thrift/src/thrift/transport/TSSLSocket.h:16:#include <zephyr/posix/sys/socket.h>
./modules/thrift/src/thrift/transport/TSSLSocket.cpp:14:#include <zephyr/posix/arpa/inet.h>
./modules/thrift/src/thrift/transport/TSSLServerSocket.cpp:9:#include <zephyr/posix/sys/socket.h>
./modules/thrift/src/thrift/transport/TSSLServerSocket.cpp:10:#include <zephyr/posix/unistd.h>
./modules/thrift/src/thrift/transport/TServerSocket.h:18:#include <zephyr/posix/sys/socket.h>
./modules/thrift/src/thrift/transport/TServerSocket.h:21:#include <zephyr/posix/netdb.h>
./modules/thrift/src/thrift/server/TFDServer.cpp:11:#include <zephyr/posix/poll.h>
./modules/thrift/src/thrift/server/TFDServer.cpp:12:#include <zephyr/posix/sys/eventfd.h>
./modules/thrift/src/thrift/server/TFDServer.cpp:13:#include <zephyr/posix/unistd.h>
```

Reply:
Currently, the standard include path is only available with `CONFIG_POSIX_API`.

Some people might want only `getopt()` for example without "the POSIX API", so those applications would still need to use the non-standard include path.

Reply:
Eventually, it would be nice to use standard include paths everywhere, but I think a big part of it was that in certain companies with large monorepos, there were header conflicts.